deps/spidershim/src/v8string.cc

```diff
@@ -96,27 +95,40 @@ MaybeLocal<String> String::NewFromUtf8(Isolate* isolate, const char* data,
   if (!twoByteChars) {
     return MaybeLocal<String>();
   }
-  JS::RootedString str(cx, JS_NewUCString(cx, twoByteChars.release(), twoByteLen));
+
+  JSString* str =
+    type == v8::NewStringType::kInternalized ?
+      JS_AtomizeAndPinUCString(cx, reinterpret_cast<const char16_t*>(twoByteChars.release())) :
```

Comment:
Erm, except that the fix is just for String::NewFromOneByte and String::NewFromTwoByte. String::NewFromUtf8 isn't susceptible to the bug, since it calls JS::UTF8CharsToNewTwoByteCharsZ, passing it the length, to get an already-truncated JS::UniqueTwoByteChars that it can pass to JS_AtomizeAndPinUCString without having to specify a length.